### ISSUE TITLE
`gradleScript` setting not working

### ISSUE THREAD

**Issue by ckipp01** [scalameta/metals#4036]
### Describe the bug

I don't have a clear reproduction yet, but just carrying this over from #vim-users on Discord where a person was setting this, but it didn't seem to be taking affect. I've never actually tested this out as I never use gradle, but it seemed to not be working in their example. This needs to be checked.

### Operating system

macOS

### Version of Metals

latest

### Commit of nvim-metals

latest

**Comment by ckipp01** [comment]
I'm transferring this over to the main repo. I tried importing this build via VS Code and also get the same issues, so it's not something specific to `nvim-metals`. Just a note about all the other stuff that was mentioned, that is all taken care of, it was due to two different metals being attached at the same time. So the main issue at this point is just that the build import in the linked repo doesn't work.

**Comment by ckipp01** [comment]
@Arthurm1 teach me obi wan, how did you get that info? I ran with `--info` and I get close, `--trace` gave me more, but still not what I wanted. I tried `--debug` but that was nuts and I still didn't see it.

**Comment by fiatjaf** [comment]
It's JDK11, yes.

That SNAPSHOT thing is published locally, but it should work if you replace it for `com.fiatjaf:immortan_2.13:0.7.0`.

**Comment by Arthurm1** [comment]
@ckipp01 I often use the `--stacktrace` option e.g. `./gradlew bloopInstall --stacktrace` to get the detail but I didn't need it here.

**Comment by Arthurm1** [comment]
@fiatjaf What's the current error you're getting when trying `./gradlew bloopInstall`?

**Comment by ckipp01** [comment]
> @ckipp01 I often use the `--stacktrace` option e.g. `./gradlew bloopInstall --stacktrace` to get the detail but I didn't need it here.

Super, thanks for the detailed explanation!

**Comment by Arthurm1** [comment]
@fiatjaf Yes - I can see that now.  All compilation failures.  So the import is successful (to a point)

You're using `apply plugin: "scala.android"`?

The `Bloop` plugin doesn't support that, only the `Scala` and `Java` plugins.  So no Scala library or compiler info is being added to the Bloop files, hence it''s struggling to compile the Scala files.

I can have a look at adding support - not sure how difficult it will be - I've not seen this plugin before.

**Comment by Arthurm1** [comment]
@fiatjaf Is this plugin still supported?  The last commit was 6 years ago and the supported Scala and Gradle versions are very old...
![image](https://user-images.githubusercontent.com/5612295/173875373-e878d597-1eb5-49ff-b883-97edc968c3bd.png)

**Comment by Arthurm1** [comment]
@fiatjaf Which dependency is the `com.samourai.wallet.schnorr.Schnorr` class coming from?

**Comment by fiatjaf** [comment]
This is the plugin I'm using: https://github.com/dsvdsv/scala-android-plugin

I don't think there are many people using it though, probably none, but compilation works fine. Maybe there is a way for me to hardcode something on the Metals/Bloop side to make it recognize things even if it doesn't understand the plugin?

I was actually thinking about making a `build.sbt` file with the dependencies just so Metals would work without actually trying to resolve the Android stuff etc.

**Comment by fiatjaf** [comment]
Sorry for taking so long to reply, there are way too many things happening here today.

That `com.samourai.wallet` thing is not declared anywhere. I removed it since it wasn't being used, but now that you mention I see it being used in a test. I should remove that test since it's not needed.

**Comment by fiatjaf** [comment]
Awesome, @Arthurm1, this worked.

For future reference, I've copied the `"scala" section from another project I had lying here into inside `.bloop/app-debug.json` (`under "project"`) and it started working.

**Comment by fiatjaf** [comment]
Weirdly, I can navigate to external dependencies and I can get the type of any variable, and even Android classes are recognized, but I can't navigate internally.

If click to go to _references_ the Metals logs output this (but nothing else happens):

```
2022.06.16 08:47:55 INFO  compiling app-debug (30 scala sources and 361 java sources)
2022.06.16 08:47:57 INFO  time: compiled app-debug in 2.28s
```

And if I click to go to _definitions_ I get this log line:

```
2022.06.16 08:47:46 ERROR code navigation does not work for the file '/home/fiatjaf/comp/obw/app/src/main/scala/fiatjaf/wallet/WalletApp.scala' because the SemanticDB file '/home/fiatjaf/comp/obw/.bloop/app-debug/build/bloop-bsp-clients-classes/classes-Metals-YY4Dbl9ASMmhAl9bzFoRyw==/META-INF/semanticdb/app/src/main/scala/fiatjaf/wallet/WalletApp.scala.semanticdb' doesn't exist. There can be many reasons for this error.
```

**Comment by Arthurm1** [comment]
@fiatjaf It works for me.

If the semantic file isn't there then it hasn't compiled properly.  Probably because it thinks that it's already compiled.

You could try the Metals `Recompile workspace` command.

If that doesn't work - you could try deleting the directories under `.bloop` - NOT the `.json` files directly under it though.  Then run the Metals `restart build server` command.

If neither of these work then it's probably down to the build export which we've hacked and you'll have to wait for Bloop to support that plugin
